test/edu/stanford/dlss/was/TestWasapiDownloader_PowerMock.java

```diff
@@ -0,0 +1,234 @@
+    Mockito.when(mockConn.pagedJsonQuery(anyString())).thenReturn(null);
+    WasapiDownloader downloaderSpy = PowerMockito.spy(new WasapiDownloader(WasapiDownloader.SETTINGS_FILE_LOCATION, args));
+    PowerMockito.doReturn(mockConn).when(downloaderSpy).getWasapiConn();
+    PowerMockito.whenNew(WasapiDownloader.class).withAnyArguments().thenReturn(downloaderSpy);
+
+    WasapiDownloader.main(args);
+    verify(mockConn).pagedJsonQuery(ArgumentMatchers.contains("collection=123"));
+    verify(mockConn, Mockito.never()).pagedJsonQuery(ArgumentMatchers.contains("crawl="));
+    verify(mockConn, Mockito.never()).pagedJsonQuery(ArgumentMatchers.contains("crawl-start-after="));
+    verify(mockConn, Mockito.never()).pagedJsonQuery(ArgumentMatchers.contains("crawl-start-before="));
+  }
+
+  @Test
+  public void main_singleFileDownload_onlyUsesFilename() throws Exception {
+    String[] args = {"--collectionId", "123", "--filename", "ARCHIVEIT-5425-MONTHLY-JOB302671-20170526114117181-00049.warc.gz" };
+    WasapiConnection mockConn = Mockito.mock(WasapiConnection.class);
+    Mockito.when(mockConn.pagedJsonQuery(anyString())).thenReturn(null);
+    WasapiDownloader downloaderSpy = PowerMockito.spy(new WasapiDownloader(WasapiDownloader.SETTINGS_FILE_LOCATION, args));
+    PowerMockito.doReturn(mockConn).when(downloaderSpy).getWasapiConn();
+    PowerMockito.whenNew(WasapiDownloader.class).withAnyArguments().thenReturn(downloaderSpy);
+
+    WasapiDownloader.main(args);
+    verify(mockConn).pagedJsonQuery(ArgumentMatchers.contains("filename=ARCHIVEIT-5425-MONTHLY-JOB302671-20170526114117181-00049.warc.gz"));
+    verify(mockConn, Mockito.never()).pagedJsonQuery(ArgumentMatchers.contains("crawl="));
+    verify(mockConn, Mockito.never()).pagedJsonQuery(ArgumentMatchers.contains("crawl-start-after="));
+    verify(mockConn, Mockito.never()).pagedJsonQuery(ArgumentMatchers.contains("crawl-start-before="));
+    verify(mockConn, Mockito.never()).pagedJsonQuery(ArgumentMatchers.contains("collection="));
+  }
+
+  private List<WasapiResponse> getWasapiRespList() {
+    List<WasapiResponse> wasapiRespList = new ArrayList<WasapiResponse>();
+    wasapiRespList.add(new WasapiResponse());
+    return wasapiRespList;
+  }
+
+  @Test
+  public void downloadSelectedWarcs_usesCrawlSelector() throws Exception {
+    WasapiConnection mockConn = Mockito.mock(WasapiConnection.class);
+    List<WasapiResponse> wasapiRespList = getWasapiRespList();
+    Mockito.when(mockConn.pagedJsonQuery(anyString())).thenReturn(wasapiRespList);
+
+    WasapiCrawlSelector mockCrawlSelector = PowerMockito.mock(WasapiCrawlSelector.class);
+    List<Integer> desiredCrawlIds = new ArrayList<Integer>();
+    desiredCrawlIds.add(Integer.valueOf("666"));
+    PowerMockito.when(mockCrawlSelector.getSelectedCrawlIds(0)).thenReturn(desiredCrawlIds);
+    PowerMockito.whenNew(WasapiCrawlSelector.class).withArguments(wasapiRespList).thenReturn(mockCrawlSelector);
```

Comment:
yes - great idea.  will do.  There's a lot about the tests I don't like right now, but shameless green and faster won the day.